pypeit/scripts/ql_keck_mosfire.py

```diff
@@ -234,104 +341,53 @@ def main(args):
     slits.mask = slits.mask_init.copy()
     # Read in the wv_calib
     wv_calib = wavecalib.WaveCalib.from_file(wvcalib_masterframe_name)
-    wv_calib.is_synced(slits)
+    #wv_calib.is_synced(slits)
     slits.mask_wvcalib(wv_calib)
     # Read in the tilts
     tilts_obj = wavetilts.WaveTilts.from_file(tilts_masterframe_name)
     tilts_obj.is_synced(slits)
     slits.mask_wavetilts(tilts_obj)
 
-
-    # Build Science image
-    sciImg = buildimage.buildimage_fromlist(spectrograph, det, parset['scienceframe'], A_files, bpm=msbpm,
-                                            slits=slits, ignore_saturation=False)
-
-    # Background Image?
-    sciImg = sciImg.sub(buildimage.buildimage_fromlist(spectrograph, det,  parset['scienceframe'], B_files, bpm=msbpm,
-                                                       slits=slits,ignore_saturation=False), parset['scienceframe']['process'])
     # Build the Calibrate object
     caliBrate = calibrations.Calibrations(None, parset['calibrations'], spectrograph, None)
     caliBrate.slits = slits
+    caliBrate.msbpm = msbpm
     caliBrate.wavetilts = tilts_obj
     caliBrate.wv_calib = wv_calib
 
-    # Instantiate Reduce object
-    # Required for pypeline specific object
-    # At instantiaton, the fullmask in self.sciImg is modified
-    redux = reduce.Reduce.get_instance(sciImg, spectrograph, parset, caliBrate, 'science', ir_redux=True, show=args.show,
-                                       det=det, std_outfile=std_outfile)
-
-    manual_extract_dict = None
-    skymodel, objmodel, ivarmodel, outmask, sobjs, waveImg, tilts = redux.run(
-        std_trace=std_trace, return_negative=True, manual_extract_dict=manual_extract_dict, show_peaks=args.show)
-
-    # TODO -- Do this upstream
-    # Tack on detector
-    for sobj in sobjs:
-        sobj.DETECTOR = sciImg.detector
+    # Find the unique throw absolute value, which defines each MASK_NOD seqeunce
+    uniq_offsets, _ = np.unique(offset_arcsec, return_inverse=True)
+    uniq_throws, uni_indx = np.unique(np.abs(offset_arcsec), return_inverse=True)
+    nuniq = uniq_throws.size
+    spec2d_list =[]
+    offset_ref = offset_arcsec[0]
+    offsets_dith_pix = []
+    # Generalize to a multiple slits, doing one slit at a time?
+    islit = 0
+    for iuniq in range(nuniq):
```

Comment:
Add some inline comments.  Not sure I follow all the logic.

Reply:
Added.